### PR TITLE
Allow override of org config with repo config

### PR DESCRIPTION
**Pull request by cblecker**
This uses similar logic to what I used for the welcome plugin. Using this, we can layer config priorities.. a repo config will always be used first, then an org config, and lastly, fall through to defaults.

### CODE REVIEW

File: prow/plugins/approve/approve.go [hunk 1]
```diff
@@ -585,13 +585,24 @@ func addApprovers(approversHandler *approvers.Approvers, approveComments []*comm
 // optionsForRepo gets the plugins.Approve struct that is applicable to the indicated repo.
 func optionsForRepo(config *plugins.Configuration, org, repo string) *plugins.Approve {
 	fullName := fmt.Sprintf("%s/%s", org, repo)
+
+	// First search for repo config
+	for i := range config.Approve {
```
Comment (fejta) [inline prow/plugins/approve/approve.go]: I wonder why these two fields don't use `for i, a := range config.Approve { if a.Repos { return &a } }`

Comment (cblecker) [reply prow/plugins/approve/approve.go]: Yeah, we don't *really* care about the key here, only the value.. so I changed that. Good idea.

File: prow/plugins/approve/approve.go [hunk 2]
```diff
@@ -585,13 +585,24 @@ func addApprovers(approversHandler *approvers.Approvers, approveComments []*comm
 // optionsForRepo gets the plugins.Approve struct that is applicable to the indicated repo.
 func optionsForRepo(config *plugins.Configuration, org, repo string) *plugins.Approve {
 	fullName := fmt.Sprintf("%s/%s", org, repo)
+
+	// First search for repo config
+	for _, i := range config.Approve {
```
Comment (fejta) [inline prow/plugins/approve/approve.go]: nit, I'd consider `a` (approve) instead of `i` (index?)

File: prow/plugins/approve/approve.go [hunk 3]
```diff
@@ -585,13 +585,24 @@ func addApprovers(approversHandler *approvers.Approvers, approveComments []*comm
...
+		if !strInSlice(fullName, i.Repos) {
+			continue
+		}
+		return &i
+	}
+
+	// If you don't find anything, loop again looking for an org config
 	for i := range config.Approve {
-		if !strInSlice(org, config.Approve[i].Repos) && !strInSlice(fullName, config.Approve[i].Repos) {
+		if !strInSlice(org, i.Repos) {
```
Comment (fejta) [inline prow/plugins/approve/approve.go]: I think this also needs the `for _, i := range config.Approve`

File: prow/plugins/approve/approve.go (hunk 2 above)
Comment (cblecker) [reply prow/plugins/approve/approve.go]: you're right, although I went with the more generic `c` for both plugins as it's a config it's iterating through

**Pull request merged**